File: Implementierung/Backend/src/main/java/edu/kit/informatik/dto/userdata/rooms/PositionDto.java
package edu.kit.informatik.dto.userdata.rooms;

import edu.kit.informatik.model.userdata.rooms.Position;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.sql.Timestamp;

/**
 * Data-Transfer-Object zur Klasse {@link Position}
 *
 * @author uekai
 * @author ugqbo
 * @version 1.0
 */
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
public class PositionDto {

    private String id;
    private String userId;
    private double xCoordinate;
    private double yCoordinate;
    private double orientation;
    private Timestamp createdAt;
    private Timestamp updatedAt;
}

File: Implementierung/frontend/paschi/src/dto/userdata/rooms/PositionDto.ts
export class PositionDto {

  id: string;
  userId: string;
  createdAt: string;
  updatedAt: string;
  xcoordinate: number;
  ycoordinate: number;
  orientation: number;

  constructor(id: string, userId: string, createdAt: string, updatedAt: string, xCoordinate: number,
              yCoordinate: number, orientation: number) {
    this.id = id;
    this.userId = userId;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.xcoordinate = xCoordinate;
    this.ycoordinate = yCoordinate;
    this.orientation = orientation;
  }
}
